### DISCUSSION TITLE
ZP_MSAISDKDEMO_COMPLETION_OAI program giving Insufficient quota message [You exceeded your current quota, please check your plan and billing details]

### DISCUSSION THREAD

**Discussion by anuragchatnow** [microsoft/aisdkforsapabap#26] [Q&A]
Hello,

I am getting below error message while executing the chat completion program for Open AI. I tried by creating multiple API Keys but its not helping

You exceeded your current quota, please check your plan and billing details.

![image](https://github.com/microsoft/aisdkforsapabap/assets/90322616/c33d46f4-c0f6-48b6-b66b-f8f3e81eb8ab)

![image](https://github.com/microsoft/aisdkforsapabap/assets/90322616/f540f340-4cf0-414e-9af0-83d165cd7061)

![image](https://github.com/microsoft/aisdkforsapabap/assets/90322616/a6f8d675-f59c-49fe-af07-2f476d136293)

**Comment by gopalnair** [comment]
Hi @anuragchatnow ,
 This message is generated from your Open AI account. Chances are, you have hit the limits of your Open AI account. If you have a Plus subscription (Open AI Paid subscription), typically, the quotas are much higher. But, with a free tier, unfortunately, the quota is easily hit.

This is not an issue within AI SDK For SAP.
